src/actions/create-config.ts

```diff
@@ -0,0 +1,62 @@
+import * as nodeFS from 'node:fs';
+import * as nodePath from 'node:path';
+import ora from 'ora';
+import type { UsagePreference } from '../prompts/usage-preference';
+import { cosmiconfigSync, type CosmiconfigResult } from 'cosmiconfig';
+import type { Context } from './context';
+
+const DEFAULT_CONFIG_FILE = '.stylelintrc.json';
+
+export function findExistingConfig(): CosmiconfigResult | null {
+	const explorer = cosmiconfigSync('stylelint');
+	const result = explorer.search();
+	return result;
+}
+
+export async function createStylelintConfig(
+	context: Context,
+	usagePreference: UsagePreference,
+): Promise<void> {
+	const spinner = ora('Creating Stylelint configuration file...').start();
+
+	if (context.dryRun) {
+		spinner.info('Creating Stylelint configuration file... (skipped due to --dry-run)');
+		spinner.stop();
+		return;
+	}
+
+	const existingConfig = findExistingConfig();
+	if (existingConfig !== null) {
+		const basename = nodePath.basename(existingConfig.filepath);
+		const failureMessage =
+			basename === 'package.json'
+				? "A Stylelint configuration is already defined in your project's `package.json` file."
+				: `A Stylelint configuration file named "${basename}" already exists in this project.`;
+
+		spinner.fail(
+			`Failed to create the Stylelint configuration file.
+			\n${failureMessage} Please remove the existing configuration file and try again.`,
+		);
+		context.exit(1);
+	}
+
+	const extendsConfig =
+		usagePreference === 'errors' ? 'stylelint-config-recommended' : 'stylelint-config-standard';
+
+	const configPath = nodePath.join(context.cwd.pathname.replace(/^\/+/, ''), DEFAULT_CONFIG_FILE);
+
+	try {
+		const dir = nodePath.dirname(configPath);
+		if (!nodeFS.existsSync(dir)) {
+			nodeFS.mkdirSync(dir, { recursive: true });
+		}
+
+		nodeFS.writeFileSync(configPath, JSON.stringify({ extends: [extendsConfig] }, null, 2));
```

Comment:
off-topic

We might add a new issue to support creating a YAML file.
e.g. `.stylelintrc.yml`

Reply:
I'm glad to see we're on the same page! I've read the [Stylelint configuration documentation](https://stylelint.io/user-guide/configure/) earlier and considered adding others options while tackling this.

According to the documentation, there are 12 possible configuration formats.

I initially considered rewriting the logic to handle all these formats (and potentially ditching cosmiconfig) to give users the flexibility to choose the one that best fits their project.

That said, I think it would be amazing to group these formats into categories, similar to how the Vite CLI handles project setup. For example, users could select a category (e.g., 'JavaScript', 'JSON', 'YAML') and then choose the specific format they prefer. This could be more intuitive and user-friendly.

Reply:
We will discuss the details in the issue itself.
I would not expect to have everything at once.
i.e.
1. JSON (this PR)
2. YAML
3. CJS or ESM (with or without [`stylelint-define-config`])
4. `stylelint.config.js` flat config (after stylelint/stylelint#7408)